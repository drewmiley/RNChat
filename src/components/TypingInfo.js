import React from 'react';

export default class TypingInfo extends React.Component {
    constructor(props) {
        super(props);
        this.shouldComponentUpdate = function(nextProps, nextState) {
            return this.props.typingState !== nextProps.typingState;
        };
    }
    componentDidMount() { this.update(); }
    componentDidUpdate() { this.update(); }
    update() {
        if (this.props.typingState.get('running')) {        
            if (this.props.typingState.get('typing')) {
                setTimeout(this.props.drewSendsMessage, this.props.typingState.get('typeTime'), this.props.messageHistory);
            } else {
                setTimeout(this.props.drewStartsTyping, this.props.typingState.get('waitTime'));
            }
        }
    }
    render() {
        return <div className='chatbot-typing-info'>{(() => { if (this.props.typingState.get('typing')) {
                return <h4><span className='label label-warning'>DREW is typing a message...</span></h4>
            }})()}</div>
    }
};