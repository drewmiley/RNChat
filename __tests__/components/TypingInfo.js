import 'react-native';
import { Map } from 'immutable';
import React from 'react';
import renderer from 'react-test-renderer';

import TypingInfo from '../..//src/components/TypingInfo';

describe('TypingInfo', () => {

    describe('Rendering', () => {

        it('should match the snapshot', () => {
        	const typingState = Map({waitTime: 1000, typeTime: 1000, typing: false, running: true});
            const component = renderer.create(
                <TypingInfo typingState={typingState} />
            );            
            expect(component).toMatchSnapshot();
        });

    });

});