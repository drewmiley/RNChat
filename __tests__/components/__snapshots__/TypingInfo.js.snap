exports[`TypingInfo Rendering should match the snapshot 1`] = `<View />`;
